### ISSUE TITLE
Default "serverpod" module should have a DB namespace prefix

### ISSUE THREAD

**Issue by dkbast** [serverpod/serverpod#2059]
**Is your feature request related to a problem? Please describe.**
Tables created in a new serverpod project don't have a prefix - since all serverpod modules have serverpod as a prefix, it's easy to see what belongs together - the same would be nice for the default module.

**Describe the solution you'd like**
Use the project name or a configureable nickname as prefix.

**Describe alternatives you've considered**
Manually adding a prefix in the model description files does not seem to be the best approach, but would also work ;)

**Additional context**
Came across this when working with my database inspector and I wanted to filter the tables

**Comment by SandPod** [comment]
I would like to clarify what the ask is here.

Do you want an automatically generated table prefix added to all tables created in a project?
Or are you encountering tables that belong to a module/server that are not prefixed?

If the first assumption is correct, one consideration to have in mind is that Postgres has a max identifier length of [63 bytes](https://www.postgresql.org/docs/16/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS); this also applies to table names. So, it might not be possible to use a dynamically assigned table prefix.

A benefit of adding the prefixes yourself is that you can group tables together based on logic in your app.

**Comment by dkbast** [comment]
> A possible solution could be to have a configurable prefix that is applied to all project created tables.

Yes I think thats what I was thinking of - could even be just a 2-4 character prefix but making it obvious that these tables come from the "core app" and not from a module. 

Also from my understanding it makes sense to move functionality / features to a module and only assemble modules in the "core" serverpod application? I wasn't aware of the identifier limitation, thats probably something to include in the docs/ and or solve by combining a (short) hash of the "speaking name" from the code with the head/tail of said "speaking name" so that developers dont accidentally break the framework? 

E.g 
table name in the yaml: my_awesome_feature_with_a_long_name
prefix: sp
```
[PREFIX]_[HASH]_[TAIL]
sp_FA14C2_a_long_name
```

I think not breaking the table lenght would also be a good lint for the LSP.

**Comment by SandPod** [comment]
> I wasn't aware of the identifier limitation...

We actually already limit you to 56 chars for table names, the rest of the chars are reserved for framework use. If you define a name longer than 56 chars we give a model error. This is reflected in the LSP as well, that line should be marked red if you have it running and modify the table name of the model.

Hashing is used internally whenever we generate SQL queries with identifiers longer than 63 chars. It probably wouldn't be too difficult to implement a similar strategy for table names.

But I think the main concern comes down to the developer experience. Not knowing the real table name of a table model makes writing custom queries more complicated. Today, you still have the power to implement that naming schema if you would like, and you will be guaranteed that that is the name the table will have in the database.
